MyAwesomeProject/components/UserList.tsx

```diff
@@ -0,0 +1,118 @@
+import React from "react"
+import { Component } from 'react'
+import { StyleSheet, Text, View, SafeAreaView, FlatList, Alert, ActivityIndicator } from "react-native"
+import { getUsersPromise } from '../queries/Users'
+import { FetchResult } from "apollo-link";
+
+interface UserPresentation {
+    name: string,
+    email: string
+}
+
+interface UserListState {
+    users: UserPresentation[]
+    loadingIcon: boolean
```

Comment:
`loadingIcon` parece mais nome de imagem do que de flag. Flags costumam seguir alguns padrões de nome:

- alguns sistemas/bibliotecas usam uma única palavra que é suficiente para descrever um estado dicotômico (i.e: `dirty`, `loading`, `pristine`)
- outros usam os verbos `is` e `has` explicitando essa dicotomia (`i.e isLoading`, `hasFailed`, `isDirty`).

É  até possível ter uma mistura dos dois, mas o importante é manter um padrão consistente (i.e `loading` para todos os estados de loading, `hasFailed` para todos estados de erro, etc).